tests/test_aurora.py

```diff
@@ -0,0 +1,159 @@
+    
+    # Convert to grid indices
+    lat_indices = np.searchsorted(grid_lat, sample_points[:, 0])
+    lon_indices = np.searchsorted(grid_lon, sample_points[:, 1])
+    grid_points = np.column_stack((lat_indices, lon_indices))
+
+    print(f"Grid points (lat, lon) indices: {grid_points}")
+
+    # Verify indices
+    assert np.all(grid_points[:, 0] < len(grid_lat))
+    assert np.all(grid_points[:, 1] < len(grid_lon))
+
+    # Process coordinates
+    pos_x, pos_y = integration_layer.preprocess_coordinates(grid_points.tolist())
+    print("Coordinate shapes - pos_x:", pos_x.shape, "pos_y:", pos_y.shape)
+
+    # Reshape coordinates for interpolation
+    pos_x = pos_x.view(batch_size, -1)
+    pos_y = pos_y.view(batch_size, -1)
+
+    # Create features for interpolation with correct size
+    features = torch.randn(batch_size, height * width * depth, output_features_dim, device=device)
+    print("Features shape for interpolation:", features.shape)
+
+    # Ensure tensors are on the correct device
+    pos_x = pos_x.to(device)
+    pos_y = pos_y.to(device)
+
+    # Perform interpolation
+    try:
+        interpolated_features = integration_layer.interpolate_features(
+            x=features,
+            pos_x=pos_x,
+            pos_y=pos_y,
+            k=1,
+            weighted=False
+        )
+        print("Interpolated features shape:", interpolated_features.shape)
+    except Exception as e:
+        print(f"Error during interpolation: {e}")
+        return
+
+    # Reshape interpolated features to match decoder input requirements
+    # The size should be [batch_size, output_features_dim, depth, height, width]
+    decoder_input = interpolated_features.view(batch_size, output_features_dim, depth, height, width)
+    print("Decoder input shape:", decoder_input.shape)
+
+    # Initialize and run decoder
+    decoder = Decoder3D(output_channels=1, embed_dim=output_features_dim, target_shape=(depth, height, width)).to(device)
+    reconstructed_output = decoder(decoder_input)
+    print("Decoder3D output shape:", reconstructed_output.shape)
+
+    # Test Fengwu_GHR functionality
+    if fengwu_ghr_config:
+        print("Testing Fengwu_GHR-specific functionality:")
+        fengwu_model = integration_layer.fengwu_model
+        if fengwu_model:
+            dummy_image_input = torch.randn(batch_size, fengwu_ghr_config.channels, 
+                                          fengwu_ghr_config.image_size[0], 
+                                          fengwu_ghr_config.image_size[1]).to(device)
+            fengwu_output = fengwu_model(dummy_image_input)
+            print("Fengwu_GHR model output shape:", fengwu_output.shape)
+
+    # Clear memory
+    del input_tensor, encoded_features, processed_latents, lora_output, reconstructed_output
+    if torch.cuda.is_available():
+        torch.cuda.empty_cache()
+
+    print("Integration test completed successfully.")
+
+if __name__ == "__main__":
```

Comment:
This shouldn't be necessary, these tests should run with `pytest` so I would remove this.